packages/user/src/views/SignUpFirstUser.tsx

```diff
@@ -0,0 +1,132 @@
+import { useTranslation } from "@dzangolab/react-i18n";
+import { Page } from "@dzangolab/react-ui";
+import { Card } from "primereact/card";
+import { useEffect, useState } from "react";
+import { toast } from "react-toastify";
+import { Link, useNavigate } from "react-router-dom";
+
+import { getIsFirstUser, signUpFirstUser } from "@/api/user";
+import SignupForm from "@/components/SignupForm";
+import { ROUTES } from "@/constants";
+import { useConfig, useUser } from "@/hooks";
+
+import { login } from "..";
+
+import type { LoginCredentials } from "@/types";
+
+const SignUpFirstUser = () => {
+  const { t } = useTranslation("user");
+
+  const appConfig = useConfig();
+  const { setUser } = useUser();
+
+  const [loading, setLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isFirstUser, setIsFirstUser] = useState<boolean>(false);
+
+  const [signUpFirstUserLoading, setSignUpFirstUserLoading] = useState(false);
+  const [loginLoading, setLoginLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getIsFirstUser(appConfig?.apiBaseUrl || "")
+      .then((response) => {
+        setIsFirstUser(response.signUp);
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const handleSubmit = (credentials: LoginCredentials) => {
+    if (!isFirstUser) {
+      return;
+    }
+
+    setSignUpFirstUserLoading(true);
+
+    signUpFirstUser(credentials, appConfig?.apiBaseUrl || "")
+      .then((response) => {
+        setSignUpFirstUserLoading(false);
+        setLoginLoading(true);
+        toast.success(`${t("firstUser.signup.messages.success")}`);
+
+        // TODO Sign up first-user should return authenticated user from api
+        login(credentials)
+          .then((result: any) => {
+            if (result?.user) {
+              setUser(result.user);
+              toast.success(`${t("login.messages.success")}`);
+            }
+          })
+          .catch((error: any) => {
+            const errorMessage = t("firstUser.login.messages.error");
+
+            if (error.name) {
+              throw error as Error;
+            }
```

Comment:
Any special reason for throwing error here? Since we want to redirect user to login page if login fails due to any reason, throwing error here block from going to next navigate statement.